app/models/recipe.rb

```diff
@@ -1,3 +1,12 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_cost
+      x = self.ingredients.map do |ing|
+         ing.cost
```

Comment:
Whenever you find yourself doing .map in a model method, there's usually an AR method that you can use instead. This is still new, so it makes sense to lean back on Ruby methods, but as much as you can, try to explore AR methods instead and avoid Ruby. 

In this case, you could do `self.ingredients.sum(:cost)`

Reply:
Nice work encapsulating this logic at the model level though! Very MVC-compliant